cmd/bucket-handlers.go

```diff
@@ -768,8 +768,10 @@ func (api objectAPIHandlers) PostPolicyBucketHandler(w http.ResponseWriter, r *h
 	pReader := NewPutObjReader(rawReader, nil, nil)
 	var objectEncryptionKey []byte
 
+	// Check if bucket encryption is enabled
+	_, encEnabled := globalBucketSSEConfigSys.Get(bucket)
 	// This request header needs to be set prior to setting ObjectOptions
-	if globalAutoEncryption && !crypto.SSEC.IsRequested(r.Header) {
+	if globalAutoEncryption || encEnabled && !crypto.SSEC.IsRequested(r.Header) {
```

Comment:
```suggestion
      if (globalAutoEncryption || encEnabled) && !crypto.SSEC.IsRequested(r.Header) {
```

otherwise if globalAutoEncryption is set and ssec headers are also set,SSE-S3 header will be added to request  incorrectly